### PR TITLE
Added event for when loading image fails + documentation

### PR DESCRIPTION
**Pull request by Saulzi**
Added event for when loading image fails on ImageOverlay layer and added missing documentation for load event

### CODE REVIEW

**Comment by Saulzi** [comment]
This is the same as PR #5415 without the extra documentation file.

File: spec/suites/layer/ImageOverlaySpec.js [hunk 1]
```diff
@@ -14,4 +14,67 @@ describe('ImageOverlay', function () {
...
+		function raiseImageEvent(event) {
+			var domEvent = document.createEvent('Event');
+			domEvent.initEvent(event);
+			overlay._image.dispatchEvent(domEvent);
+		}
+
+		describe('when loaded', function () {
+			it('should raise the load event', function () {
+				var loadraised = false;
+				overlay.once('load', function () { loadraised = true; });
```
Comment (perliedman) [inline spec/suites/layer/ImageOverlaySpec.js]: While this works, you can also use `sinon` to do this and similar tasks in a more explicit fashion, see for example https://github.com/Leaflet/Leaflet/blob/master/spec/suites/core/EventsSpec.js#L20

File: spec/suites/layer/ImageOverlaySpec.js [hunk 2]
```diff
@@ -14,4 +14,67 @@ describe('ImageOverlay', function () {
...
+				raiseImageEvent('load');
+				expect(loadraised).to.be(true);
+			});
+		});
+
+		describe('when load fails', function () {
+			it('should raise the error event', function () {
+				var errorRaised = false;
+				overlay.once('error', function () { errorRaised = true; });
```
Comment (perliedman) [inline spec/suites/layer/ImageOverlaySpec.js]: ...same here, `sinon` would be great.

Comment (Saulzi) [inline spec/suites/layer/ImageOverlaySpec.js]: So basically the same as jasmine as you can do jasmine.spy. 

Will bear that in mind for any future contributions :)

**Pull request merged**